src/hooks/use-poll.js

```diff
@@ -9,23 +9,46 @@ const useConst = (factory) => {
     return ref.current
 }
 
-const useLazyInterval = (fn, interval) => {
+export const useLazyInterval = (fn, interval) => {
```

Comment:
(a summary for `usePoll` also would be nice)

Reply:
Note that I didn't add this code, just reused and fixed some problems with it. Previously this would keep polling if the request took longer than the interval, so I refactored it to not poll until after the promise is resolved.

I think `lazy` refers to the fact you have to start it manually. Like passing lazy to `useDataQuery`. Its not started on mount.

I can add some docs.

I tried to use usePoll, but didn't have access to all info for cancellation in the callback. Also needed more control over the query options etc. So I decided to just reuse `useLazyInterval`